### PR TITLE
Adjustments to metaschema data types to align with Commonmark.

### PR DESCRIPTION
**Pull request by david-waltermire-nist**
# Committer Notes

This PR provides significant support for additional HTML tags in the `markup-line` and `markup-multiline` implementations.

Support is added for:
- `<hr/>`
- `<blockquote>`
- `<br>`
- `<code class="someclass">`

Titles are now allowed on `<a title="sometitle">` and `` tags. This was a major oversight.

List items (`<li>`) are now more robust in the content they allow. They now allow `<pre>`, `<hr/>`, and `<blockquote>`, headings, and `<p>` tags.

These schema changes support #70, #109, and #221.
 

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/metaschema/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/metaschema/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [x] Have you updated all website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes you made? Changes to the website can be made in the website/content directory of your branch.

### CODE REVIEW

File: test-suite/schema-generation/allowed-values/allowed-values-basic_test_valid_PASS.json [hunk 1]
```diff
@@ -3,16 +3,21 @@
...
                 "field-date": "2020-01-01"
             },
             {
-                "even-flag": 2,
+                "even-flag": 10,
                 "field-date": "2020-02-29"
+            },
+            {
+                "even-flag": 2,
+                "field-date": "2020-07-04"
```
Comment (aj-stein-nist) [inline test-suite/schema-generation/allowed-values/allowed-values-basic_test_valid_PASS.json:20]: Minor, non-blocking: I just want to make sure I understand: these changes are minor and unrelated to the testing of the PR, correct? I didn't know if there was some relation in the test suite to these changes around Markdown/CommonMark conformance?

File: schema/xml/metaschema-markup-multiline.xsd [hunk 2]
```diff
@@ -92,4 +125,11 @@
       <xs:enumeration value="right"/>
     </xs:restriction>
   </xs:simpleType>
+
+  <xs:complexType name="blockQuoteType">
+    <!-- TODO: confirm if nested blockquote is supported in markdown -->
```
Comment (aj-stein-nist) [inline schema/xml/metaschema-markup-multiline.xsd:130]: Minor, non-blocking: do we mean Markdown generally or specifically Markdown? It seems the original description (I could jokingly call it a specification 😆) did support it, and [CommonMark supports it and tests for it](https://spec.commonmark.org/dingus/?text=%3E%20deeply%20nested%20blockquote%0A%3E%3E%20deeply%20nested%20blockquote%0A%3E%3E%3E%3E%20deeply%20nested%20blockquote%0A%3E%3E%3E%3E%3E%20deeply%20nested%20blockquote%0A%3E%3E%3E%3E%3E%3E%20deeply%20nested%20blockquote). Is this what you're looking for? (This example is from the commonmark.js test suite itself.)

**Review by aj-stein-nist — APPROVED** [review]

File: schema/xml/metaschema-markup-multiline.xsd (hunk 2 above)
Comment (wendellpiez) [reply schema/xml/metaschema-markup-multiline.xsd:130]: Indeed, it's a big help. :+1:

Indeed this is probably true of any child of HTML `blockquote` especially lists, for reasons not unrelated to list nesting.

If something is hard to parse and we mandate it, it is one thing if CommonMark already does it, and another if no one really does it, or no one does it testably and consistently. (Unfortunately, giving one simple example and saying 'we do this' doesn't actually cut it since it's no real help to show only the happy path.)

Comment (aj-stein-nist) [reply schema/xml/metaschema-markup-multiline.xsd:130]: Well I was asking because CommonMark is what we are aiming at conformance with (sorry I wrote Markdown twice 🤦), it seems we have confirmed that CommonMark allows nested blockquotes and quickly testing one of their reference implementations backs this, no?

File: test-suite/schema-generation/allowed-values/allowed-values-basic_test_valid_PASS.json (hunk 1 above)
Comment (david-waltermire-nist) [reply test-suite/schema-generation/allowed-values/allowed-values-basic_test_valid_PASS.json:20]: Yes. This was just a unit test improvement to add some diversity in the test vector values..

**Review by wendellpiez — APPROVED** [review]
LGTM, thanks Dave.

**Review by wendellpiez — APPROVED** [review]
LGTM
